server/modules/taxonomy/api/taxonomyApi.js

```diff
@@ -147,13 +151,18 @@ export const init = app => {
       try {
         const { surveyId, taxonomyUuid, draft } = Request.getParams(req)
 
-        Response.setContentTypeFile(res, `taxonomy_${taxonomyUuid}.csv`, null, Response.contentTypes.csv)
+        const survey = await SurveyService.fetchSurveyById(surveyId, draft)
+        const surveyName = Survey.getName(Survey.getSurveyInfo(survey))
+        const taxonomy = await TaxonomyService.fetchTaxonomyByUuid(surveyId, taxonomyUuid, draft)
```

Comment:
survey and taxonomy can be fetched using Promise.all

Reply:
right